Session/Home/HomeViewModel.swift

```diff
@@ -578,16 +617,44 @@ public class HomeViewModel: NavigatableStateHolder {
         dependencies[defaults: .standard, key: .rateAppRetryAttemptCount] = 0
         
         if let scene = UIApplication.shared.connectedScenes.first(where: { $0.activationState == .foregroundActive }) as? UIWindowScene {
-            SKStoreReviewController.requestReview(in: scene)
+            NotificationCenter.default.addObserver(
+                self,
+                selector: #selector(self.systemHasPresentedNewWindow(notification:)),
+                name: UIWindow.didBecomeVisibleNotification, object: nil
+            )
+            
+            if !dependencies[feature: .simulateAppReviewLimit] {
+                SKStoreReviewController.requestReview(in: scene)
+            }
+
+            // Added 2 sec delay to give time for requet review to proc
+            DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(2)) { [self] in
```

Comment:
Please change this to a `weak self` capture - it's not an issue with the current code but capturing `self` strongly means if there were future changes to the code it'd be easier to miss a retain cycle (you can either `guard let self = self else { return }` of `NotificationCenter.default.removeObserver(self as Any, ...` to resolve the warning that this creates)